hasjob/models/filterset.py

```diff
@@ -0,0 +1,159 @@
+    tags = db.relationship(Tag, secondary=filterset_tag_table)
+    domains = db.relationship(Domain, secondary=filterset_domain_table)
+    location_geonameids = db.Column(postgresql.ARRAY(db.Integer(), dimensions=1), nullable=True, index=True)
+    remote_location = db.Column(db.Boolean, default=False, nullable=False, index=True)
+    pay_currency = db.Column(db.CHAR(3), nullable=True, index=True)
+    pay_cash_min = db.Column(db.Integer, nullable=True, index=True)
+    equity = db.Column(db.Boolean, nullable=False, default=False, index=True)
+    keywords = db.Column(db.Unicode(250), nullable=False, default=u'', index=True)
+
+    def __repr__(self):
+        return '<FilterSet %s "%s">' % (self.board.title, self.title)
+
+    @classmethod
+    def get(cls, board, name):
+        return cls.query.filter(cls.board == board, cls.name == name).first_or_404()
+
+    def url_for(self, action='view', _external=False, **kwargs):
+        if action == 'view':
+            subdomain = self.board.name if self.board.not_root else None
+            return url_for('filterset_view', subdomain=subdomain, name=self.name, _external=_external)
+
+    def to_filters(self, translate_geonameids=False):
+        from hasjob.views.helper import location_geodata
+
+        location_names = []
+        if translate_geonameids and self.location_geonameids:
+            location_dict = location_geodata(self.location_geonameids)
+            for geonameid in self.location_geonameids:
+                location_names.append(location_dict[geonameid]['name'])
+
+        return {
+            't': [jobtype.name for jobtype in self.types],
+            'c': [jobcategory.name for jobcategory in self.categories],
+            'l': location_names if translate_geonameids else self.location_geonameids,
+            'currency': self.pay_currency,
+            'pay': self.pay_cash_min,
+            'equity': self.equity,
+            'anywhere': self.remote_location
+        }
+
+    @classmethod
+    def from_filters(cls, board, filters):
+        basequery = cls.query.filter(cls.board == board)
+
+        if filters.get('t'):
+            basequery = basequery.join(
+                filterset_jobtype_table).join(
+                JobType).filter(JobType.name.in_(filters['t']))
+        else:
+            basequery = basequery.filter(~cls.id.in_(db.select([filterset_jobtype_table.c.filterset_id])))
+
+        if filters.get('c'):
+            basequery = basequery.join(
+                filterset_jobcategory_table).join(
+                JobCategory).filter(JobCategory.name.in_(filters['c']))
+        else:
+            basequery = basequery.filter(~cls.id.in_(db.select([filterset_jobcategory_table.c.filterset_id])))
+
+        if filters.get('l'):
+            basequery = basequery.filter(cls.location_geonameids == sorted(filters['l']))
+        else:
+            basequery = basequery.filter(cls.location_geonameids == None)
```

Comment:
Will `== None` match an empty array? Also, it'll need a `# NOQA` for PyFlakes.

Reply:
Thanks, changed this to `[]` (defaults to `[]`).